### PR TITLE
Adding 3d Pooling using latest MIOpen API

### PR DESCRIPTION
**Pull request by jerryyin**
This review pull in the latest MIOpen API in 3d pooling. The MIOpen dependency should have the following two PRs: 1. [#2165](https://github.com/AMDComputeLibraries/MLOpen/pull/2165) 3D Pooling Part 1/2: API, driver and ctest refactor for Nd cases 2. [#2210](https://github.com/AMDComputeLibraries/MLOpen/pull/2210) 3D Pooling Part 2/2: kernels and ctest;

Using the updated MIOpen API, I am able to set 3d pooling and pass the follow unit test:
`//tensorflow/python/kernel_tests:pooling_ops_3d_test`

**TODO**

- When CI is back and autobuilds docker is ready, I need to rebuild latest MIOpen using the latest 3.0 ROCm infrastructure and include it as the docker dependency.

- Will need to re-enable 3d pooling related tests

### CODE REVIEW

**Comment by jerryyin** [comment]
Adding [DO NOT MERGE] tag, will need to: 

- finish rest of todo list; 
- do a full CI regression and 
- close out issue deven mentioned.

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

File: tensorflow/stream_executor/rocm/rocm_dnn.cc [hunk 1]
```diff
@@ -188,6 +188,8 @@ namespace wrap {
   __macro(miopenCreateTensorDescriptor)                    \
   __macro(miopenDestroyTensorDescriptor)                   \
   __macro(miopenSet2dPoolingDescriptor)                    \
```
[the remark this answers is not in the document]
Comment (jerryyin) [reply tensorflow/stream_executor/rocm/rocm_dnn.cc]: Done. Just to confirm: looks like this feature includes some latest change from MIOpen thus should be put in `master-rocm-enhanced` branch?

[the remark this answers is not in the document]
Comment (deven-amd) [reply tensorflow/stream_executor/rocm/rocm_dnn.cc]: yes, it should make it into `master-rocm-enhanced` branch.

We do not yet have the pre-reqs in place for accepting checkins into `master-rocm-enhanced` branch, but you can go ahead and file the PRs (see PR #782 #783 for examples)

For the time being, check this into `develop-upstream` and create he PR for  `master-rocm-enhanced`.

[the remark this answers is not in the document]
Comment (jerryyin) [reply tensorflow/stream_executor/rocm/rocm_dnn.cc]: Thanks. Will do. Once the CI is passing, I'll submit the code change as well as filing it to `master-rocm-enhanced`.

**Comment by ekuznetsov139** [comment]
We still need MLOpen-side patches merged into the public repo (https://github.com/ROCmSoftwarePlatform/MIOpen). Otherwise tensorflow-upstream can't be built with the public MIOpen.

**Comment by jerryyin** [comment]
Thanks for reminding. This feature is targeted for `master-rocm-enhanced` (See #803 ) for now until the patch becomes public.

**Comment by ekuznetsov139** [comment]
Is there a known good tag of MLOpen I can use with this patch? I tried to run TF tests with the head of develop and got 50+ failures of the type "tensorflow/core/framework/op_kernel.cc:1675] OP_REQUIRES failed at conv_ops.cc:1179 : Not found: No algorithm worked!"

**Comment by jerryyin** [comment]
@ekuznetsov139 You need to install it using [this line](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/775/files#diff-996bfccf1a57ef3e5a9c3a010c75271aR76)

**Pull request merged**